### PR TITLE
get_activation('relu') provides a simple mapping from strings in configs to activation functions

### LINKED ISSUE #2786: SequenceSummary: config.summary_activation = 'relu' would be ignored
This isn't a bug, but merely an unintuitive argument name.

`summary_activation` sounds like it can be general, e.g. "relu" or "gelu" or something, but if its not "tanh" its ignored.

Since I assume it's annoying to go through all the configs and rename a field to use_tanh=True, I propose that we raise if summary_activation is a string thats not tanh, instead of silently just using no activation.

Another approach could integrate an `ACT2FN` dictionary (see  https://github.com/huggingface/transformers/issues/1347)
to actually support the other activation functions.

Happy to do either approach if others think it would be useful.

### PR DESCRIPTION
**Pull request by sshleifer**

### CODE REVIEW

File: src/transformers/modeling_openai.py [hunk 1]
```diff
@@ -114,15 +115,7 @@ def load_tf_weights_in_openai_gpt(model, config, openai_checkpoint_folder_path):
...
-def gelu(x):
-    return 0.5 * x * (1 + torch.tanh(math.sqrt(2 / math.pi) * (x + 0.044715 * torch.pow(x, 3))))
-
-
-def swish(x):
-    return x * torch.sigmoid(x)
-
-
-ACT_FNS = {"relu": nn.ReLU, "swish": swish, "gelu": gelu}
+ACT_FNS = {"relu": nn.ReLU, "swish": swish, "gelu": gelu_new}
```
Comment (sshleifer) [inline src/transformers/modeling_openai.py]: gelu_new is the openAI one, as you will see later.

File: src/transformers/pytorch_activations.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "gelu": gelu,
+    "tanh": F.tanh,
+    "gelu_new": gelu_new,
+}
+
+
+def get_activation(activation_string):
+    if activation_string in ACT2FN:
+        return ACT2FN[activation_string]
+    else:
```
Comment (sshleifer) [inline src/transformers/pytorch_activations.py]: considered trying to find the desired activation string in torch.nn.functional, e.g. `getattr(F, activation_string)` before raising, as they will probably add more over time and we can avoid updating, but I was scared of side effects.

File: src/transformers/pytorch_activations.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    """ Original Implementation of the gelu activation function in Google Bert repo when initially created.
+        For information: OpenAI GPT's gelu is slightly different (and gives slightly different results):
+        0.5 * x * (1 + torch.tanh(math.sqrt(2 / math.pi) * (x + 0.044715 * torch.pow(x, 3))))
+        This is now written in C in torch.nn.functional
+        Also see https://arxiv.org/abs/1606.08415
+    """
+    return x * 0.5 * (1.0 + torch.erf(x / math.sqrt(2.0)))
+
+
+gelu = getattr(F, "gelu", _gelu_python)
```
Comment (LysandreJik) [inline src/transformers/pytorch_activations.py]: That's cool

**Review by LysandreJik — APPROVED** [review]
Nice and clean

**Review by julien-c — APPROVED** [review]
Random thoughts:
- Should we do this for TF too
- Should the file name then be just `activations.py` to be consistent with our naming so far?
- Strictly speaking, we _could_ probably change some backward compatibility in the existing configs, if we really needed to (cf. your other issue related to this). For instance, it'd be obviously a lot cleaner if the `gelu` in openai was called `gelu_new` – probably not worth breaking the compatibility over, but a good thing to keep in mind for the future

Other than that, LGTM! Thanks Sam!

**Comment by sshleifer** [comment]
Happy to do TF in a separate PR. I don't think worth breaking backwards compatibility over this.

**Pull request merged**